x/gamm/twap/hook_test.go

```diff
@@ -0,0 +1,81 @@
+package twap_test
+
+import (
+	"time"
+
+	"github.com/osmosis-labs/osmosis/v10/x/gamm/twap/types"
+)
+
+// TestCreatePoolFlow tests that upon a pool being created,
+// we have made the correct store entries.
```

Comment:
I think we should also be more specific about what exactly we are testing in this file. At first, I was looking in the `hook_listener.go` due to the name of this file. However, then, I released that a lot of the functionality is actually related to `endBlock` which is defined in `logic.go`

So I suggest renaming this to `hook_functional_test.go` and adding a comment that all these tests are functional and testing the relationship between twap with other system components.